### PR TITLE
CRM-20425 - Configure inbound email activity status

### PR DESCRIPTION
**Pull request by colemanw**
* [CRM-20425: Activity status per mail account](https://issues.civicrm.org/jira/browse/CRM-20425)

### CODE REVIEW

File: templates/CRM/Admin/Form/MailSettings.tpl [hunk 1]
```diff
@@ -67,7 +67,20 @@
...
+
+  <tr class="crm-mail-settings-form-block-activity_status_id"><td class="label">{$form.activity_status_id.label}</td><td>{$form.activity_status_id.html}</td></tr>
     </table>
   <div class="crm-submit-buttons">{include file="CRM/common/formButtons.tpl" location="bottom"}</div>
 {/if}
 </div>
+{literal}
+<script type="text/javascript">
+  CRM.$(function($) {
+    var $form = $('form.{/literal}{$form.formClass}{literal}');
```
Comment (mickadoo) [inline templates/CRM/Admin/Form/MailSettings.tpl]: I'm not so good with JS, but will this function be applied to new items in the list? I just remember sometimes having issues where functions that are called on state change were not called if the element was added after page load.

**Review by mickadoo — COMMENTED** [review]
I just left a few comments, mostly questions or minor suggestions about line length.

**Comment by colemanw** [comment]
@mickadoo thanks for the review. I've taken your suggestions and changed the field to store a name instead of an id, and removed the existing hardcoded `2` from the email processor. Since we're using the api, the machine name "Completed" works equally well and is more readable/portable.

Also there were some stray unrelated changes (CRM/Report/Form/Activity.php) that accidentally snuck into this PR. I've rebased them out. This should be good to go. Can you test it on your system if you haven't already done so to verify it correctly sets the activity status of inbound emails?

File: templates/CRM/Admin/Form/MailSettings.tpl (hunk 1 above)
Comment (colemanw) [inline templates/CRM/Admin/Form/MailSettings.tpl]: Yes it works in both add & edit.

File: CRM/Admin/Form/MailSettings.php [hunk 2]
```diff
@@ -94,6 +95,26 @@ public function addRules() {
...
+   * Add local and global form rules.
+   */
+  public function setDefaultValues() {
+    $defaults = parent::setDefaultValues();
+
+    // Set activity status to "Completed" by default.
+    if ($this->_action != CRM_Core_Action::DELETE &&
+      (!$this->_id || !CRM_Core_DAO::getFieldValue('CRM_Core_BAO_MailSettings', $this->_id, 'activity_status'))
+    ) {
+      $defaults['activity_status'] = 'Completed';
```
Comment (mickadoo) [inline CRM/Admin/Form/MailSettings.php]: Just to check if I understand this conditional:

IF (this is not a deletion) AND ((this is a creation) OR (there is no default value for activity status)) THEN status should be "Completed"

Does that sound right?

**Comment by mickadoo** [comment]
@colemanw Looks good to me.

About testing locally, I'm really not so familiar with using core CiviCRM, except what I see of it through CiviHR. To my great shame I don't even know how to test your changes locally. I have a container I was using to try run the core tests and from there I ran `civibuild install` according to the documentation [here](https://buildkit.civicrm.org/#/tutorials). I then went to the civicrm folder (`/opt/buildkit/build/mytestbuild/sites/all/modules/civicrm/`) and added your fork as a git remote and checked out your branch. I couldn't see any of the menus though and in general the site looks a bit messed up so I'm pretty sure I'm doing something wrong.

As regards the review, it's fine for me. If you'd like me to test locally I might need some help getting set up though.

**Review by mickadoo — APPROVED** [review]
looks fine, thanks for splitting it

**Pull request merged**